File: akka-samples/akka-sample-camel/src/main/scala/Producer1.scala
package sample.camel

import se.scalablesolutions.akka.actor.Actor
import se.scalablesolutions.akka.camel.Producer

/**
 * @author Martin Krasser
 */
class Producer1 extends Actor with Producer {

  def endpointUri = "direct:welcome"

  override def oneway = false // default
  override def async = true   // default

  protected def receive = produce
}

File: akka-docs/java/code/docs/camel/sample/route/Producer1.java
package docs.camel.sample.route;
//#CustomRouteExample
import akka.camel.javaapi.UntypedProducerActor;

public class Producer1 extends UntypedProducerActor{
  public String getEndpointUri() {
    return "direct:welcome";
  }
}
//#CustomRouteExample
